### PR TITLE
Issue 128

### PR DESCRIPTION
**Pull request by jbrillon**
- Addressing issue #128 
- Cleaning up the split form related functions in physics
- correcting SURFACE_CONFORMING_TEST test name (e.g. `2SURFACE_CONFORMING_TEST` is now `2D_SURFACE_CONFORMING_TEST`)

### CODE REVIEW

File: src/physics/mhd.cpp [hunk 1]
```diff
@@ -324,42 +324,6 @@ ::convective_flux (const std::array<real,nstate> &conservative_soln) const
     return conv_flux;
 }
 
-template <int dim, int nstate, typename real>
-std::array<dealii::Tensor<1,dim,real>,nstate> MHD<dim, nstate, real>
-::convective_numerical_split_flux(const std::array<real,nstate> &soln_const,
```
Comment (PranshulThakur) [inline src/physics/mhd.cpp:329]: Is this function in MHD being replaced by something else?

File: src/physics/physics.cpp [hunk 2]
```diff
@@ -48,6 +52,27 @@ PhysicsBase<dim,nstate,real>::PhysicsBase(
         manufactured_solution_function_input)
 { }
 
+template <int dim, int nstate, typename real>
+std::array<dealii::Tensor<1,dim,real>,nstate> PhysicsBase<dim,nstate,real>::convective_numerical_split_flux (
+    const std::array<real,nstate> &/*conservative_soln1*/,
+    const std::array<real,nstate> &/*conservative_soln2*/) const
+{
+    pcout << "ERROR: convective_numerical_split_flux() has not yet been implemented. Aborting..." <<std::flush;
```
Comment (PranshulThakur) [inline src/physics/physics.cpp:60]: Hmm, if it is implemented for some PDEs, maybe something like "ERROR: convective_numerical_split_flux() has not yet been implemented for and overridden by the selected PDE. Aborting..." will be good.

**Review by PranshulThakur — COMMENTED** [review]
Great PR! I have a few editorial comments.

File: src/physics/mhd.cpp (hunk 1 above)
Comment (jbrillon) [reply src/physics/mhd.cpp:329]: it was put there as a placeholder -- MHD never had an implementation for it

File: src/physics/physics.cpp (hunk 2 above)
Comment (jbrillon) [reply src/physics/physics.cpp]: fixed

File: src/physics/euler.h [hunk 3]
```diff
@@ -277,15 +277,10 @@ class Euler : public PhysicsBase <dim, nstate, real>
     /** See the book I do like CFD, sec 4.14.2 */
     real compute_pressure_from_density_temperature ( const real density, const real temperature ) const;
 
-    /// The Euler split form is that of Kennedy & Gruber or Ismail & Roe.
+    /// The Euler split form convective flux
```
Comment (cpethrick) [inline src/physics/euler.h:280]: Add another comment here:
`/** Evaluates convective flux based on the chosen split form.`

File: src/physics/euler.cpp [hunk 4]
```diff
@@ -632,15 +632,6 @@ ::convective_numerical_split_flux_ismail_roe(const std::array<real,nstate> &cons
     return conv_num_split_flux;
 }
 
-template <int dim, int nstate, typename real>
-real Euler<dim, nstate, real>
-::convective_surface_numerical_split_flux (
```
Comment (cpethrick) [inline src/physics/euler.cpp:637]: Was this also a placeholder?

**Review by cpethrick — COMMENTED** [review]
Looks good, I prefer this solution over placeholders. I've requested one documentation change, and will approve when that's addressed!

File: src/physics/euler.h (hunk 3 above)
Comment (jbrillon) [inline src/physics/euler.h:280]:
```suggestion
    ///  Evaluates convective flux based on the chosen split form.
```

File: src/physics/euler.cpp (hunk 4 above)
Comment (jbrillon) [reply src/physics/euler.cpp:637]: I will double check with him though

Comment (jbrillon) [reply src/physics/euler.cpp:637]: It's not even being called anywhere in the code; maybe we can remove it if its not being used?

Comment (jbrillon) [reply src/physics/euler.cpp:637]: @AlexanderCicchino says its deprecated, I'm going to remove the function

Comment (cpethrick) [reply src/physics/euler.cpp:637]: Okay, sounds good!

Comment (jbrillon) [reply src/physics/euler.cpp:637]: done

Comment (cpethrick) [reply src/physics/euler.cpp:637]: sounds good!

**Review by cpethrick — APPROVED** [review]

**Pull request merged**